rxbindings/src/main/java/com/amplifyframework/rx/RxHubBinding.java

```diff
@@ -53,13 +53,8 @@
     @Override
     public Observable<HubEvent<?>> on(@NonNull HubChannel hubChannel) {
         return Observable.defer(() -> Observable.create(emitter -> {
-            final CompositeDisposable disposable = new CompositeDisposable();
-            emitter.setDisposable(disposable);
             SubscriptionToken token = hub.subscribe(hubChannel, emitter::onNext);
-            disposable.add(Disposables.fromAction(() -> {
-                hub.unsubscribe(token);
-                emitter.onComplete();
-            }));
+            emitter.setDisposable(Disposables.fromAction(() -> hub.unsubscribe(token)));
```

Comment:
Maybe I'm missing something but is this the type of thing you'd want to have as another method in your AmplifyDisposables factory?

Reply:
Ah, good idea. `Disposables.fromHubSubscription(hub, token)` ? Something like that?